code/modules/projectiles/guns/energy.dm

```diff
@@ -42,7 +42,7 @@
 	// What percentage of the full battery a shot will expend
 	var/shot_cost_percent = round(clamp(shot.e_cost / cell.maxcharge, 0, 1) * 100)
 	// Ignore this on oversized/infinite cells or ammo without cost
-	if(shot_cost_percent > 0)
+	if(shot_cost_percent >= 0)
```

Comment:
The runtime occurs because shot_cost_percent is 100, which causes max_shots to be 0 because it takes away 1 at the end for some strange reason. Changing this check to allow 0 would allow a different runtime to happen if the shot costs no energy.

A better fix would be to just butcher the shot_cost_percent, we don't really need that. Check if the shot.e_cost is above 0 (and cache that var). max_shots would just be round(cell.maxcharge / e_cost, assuming the cell requires at least e_cost energy to fire the last shot and round() still floors. shots_left would just be round(cell.charge / e_cost).

Also make pitch_to_use start at 1 instead of null in case the e_cost check doesn't pass.

Reply:
honk